purchase_order_no_zero_price/models/purchase_order_line.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (C) 2022 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, exceptions, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    @api.constrains("price_unit", "display_type", "state")
+    def _check_price_unit_zero(self):
+        # Prevent zero prices on confirmed POs
+        # to help ensure proper Stock Valuation
+        # when receiving the products
+        # Also supports `purchase_triple_discount` module
+        no_price_lines = self.filtered(
+            lambda x: not x.display_type
+            and x.state not in ["draft", "done", "cancel"]
+            and not x.price_unit
+            # Allow zero price for 100% discounts:
+            and (
+                x.discount + getattr(x, "discount2", 0.0) + getattr(x, "discount3", 0.0)
```

Comment:
if you replace : 

x.discount + getattr(x, "discount2", 0.0) + getattr(x, "discount3", 0.0)

by 
getattr(x, "discount", 0.0) + getattr(x, "discount2", 0.0) + getattr(x, "discount3", 0.0)

you can change the dependency module from ``purchase_discount`` to ``purchase`` and so make the module usable in more case.